src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -484,6 +484,76 @@ namespace shogun
 			    ->add_col_vec(A, i, b, result, alpha, beta);
 		}
 
+		/** Calculates pseudo inverse A+ from eigen values solved by a self adjoint eigen solver.
+		 * User should pass an appropriately pre-allocated memory matrix
+		 * or pass the operand argument A as a result.
+		 *
+		 * @param A The matrix
+		 * @param result The matrix that saves the result
+		 */
+		template <typename T>
+		void pinvh(const SGMatrix<T>& A, SGMatrix<T>& result)
+		{
+
+			REQUIRE(
+			    result.num_rows == A.num_rows && result.num_cols == A.num_cols,
+			    "Dimension mismatch! A (%d x %d) vs result (%d x %d).\n",
+			    A.num_rows, A.num_cols, result.num_rows, result.num_cols);
+	
+			REQUIRE(A.num_rows == A.num_cols, "Given Matrix is not square A(%d x %d)", A.num_rows, A.num_cols);
+					
+			infer_backend(A)->pinvh(A, result);
+		}
+
+		/**
+		 * Calculates pseudo inverse A+ from eigen values solved by a self adjoint eigen solver.
+		 * This version returns the result in a newly created matrix.
+		 *
+		 * @param A The matrix
+		 * @return The result matrix
+		 */
+		template <typename T>
+		SGMatrix<T> pinvh(const SGMatrix<T>& A)
+		{
+			auto result = A.clone();
+			pinvh(A, result);
+			return result;
+		}
+
+		/** Calculates pseudo inverse A+ using singular value decomposition.
+		 * User should pass an appropriately pre-allocated memory matrix
+		 * or pass the operand argument A as a result.
```

Comment:
"or pass the operand argument A as a result" mean?
I dont think that can work here as the dimension doesnt work out or? It would only work for square matrices
But you need to add a test for this as well
`linalg::pinvh(A, A)`